lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -226,11 +231,18 @@ function checkBucketPolicy(policy, requestType, canonicalID, arn, log) {
     return permission;
 }
 
-function isBucketAuthorized(bucket, requestType, canonicalID, arn, log) {
+function isBucketAuthorized(bucket, requestType, canonicalID, authInfo, log) {
     // Check to see if user is authorized to perform a
     // particular action on bucket based on ACLs.
     // TODO: Add IAM checks
-    if (bucket.getOwner() === canonicalID) {
+    let requesterIsNotUser = true;
+    let arn = null;
+    if (authInfo) {
+        requesterIsNotUser = !authInfo.isRequesterAnIAMUser();
+        arn = authInfo.getArn();
+    }
```

Comment:
minor: duplicate code(https://github.com/scality/cloudserver/pull/2354/files#diff-c3ebea7ca58e537d85230bd2a8d75ec8R270-R274)

```suggestion
const arn = _getArn(authInfo)
```